pkg/types/types.go

```diff
@@ -260,8 +261,9 @@ func MergeMetrics(metrics [][]Metric, replicaMatchMode cfg.ReplicaMatchMode, rep
 		metricsStat.DataPointCount += stats.DataPointCount
 	}
 
-	if metricsStat.MismatchCount > 0 {
-		corruptionLogger.Warn("metric replica mismatch observed",
+	metricsUnfixedMismatchCount := metricsStat.MismatchCount - metricsStat.FixedMismatchCount
+	if metricsUnfixedMismatchCount > 0 {
+		corruptionLogger.Warn("metric unfixed replica mismatch observed",
```

Comment:
@emadolsky Would you please consider refactoring-out the `corruptionLogger` in the future PRs?

Reply:
Yeah sure.